lib/backends/sdc/containers.js

```diff
@@ -3647,6 +3651,14 @@ function _runAttach(opts, callback) {
     if (!serverSocket) {
         serverSocket = net.createConnection({ host: host, port: port });
         serverSocket.on('connect', setupListeners);
+        serverSocket.on('error', function (error) {
+            opts.log.debug('attach for %s threw an error %',
+                cmdString, error.toString());
+
+            serverSocket.end();
```

Comment:
You mean serverSocket.end() is not needed? If so, I'll remove from the other call as well.